Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/Parse/Inlines/MarkdownLinkInline.cs

```diff
@@ -264,6 +272,21 @@ public void ResolveReference(MarkdownDocument document)
             ReferenceId = null;
         }
 
+        /// <summary>
+        /// Checks if the given URL is an Email.
+        /// </summary>
+        /// <param name="url"> The URL to check. </param>
+        /// <returns> <c>true</c> if the URL is valid; <c>false</c> otherwise. </returns>
+        private static bool IsUrlEmail(string url)
+        {
+            if (Regex.IsMatch(url, @"^([\w\.\-]+)@([\w\-]+)((\.(\w){2,})+)$"))
```

Comment:
I don't think the RegEx here is sufficient.

If we're on RS3, we should probably add a method on a .NET Standard 2.0 lib that exposes the MailAddress class, but not sure how we could easily check/switch for that in the main code.  @nmetulev thoughts?

Reply:
We can't use .NET standard 2.0 only APIs just yet. But we can use the C# version [from here](http://emailregex.com/), not sure if that is much different than what you already have?

Reply:
@nmetulev the regex is same as the one which was used before in `RegexTextBox` which is now depreciated. See #1192.

Reply:
Yes, but that wouldn't get an e-mail like `blah'-m$&an-1@gmail.com` which is valid according to the [spec](https://tools.ietf.org/html/rfc5322).  On the reference page from @nmetulev, the `C#` one or the full `General Email Regex (RFC 53222 Official Standard)` both should work in UWP and cover unusual addresses.

We just want to make sure since we're re-adding this sort of thing, that we're grabbing the right source.

Reply:
@michael-hawker @nmetulev will make change asap.